src/main/java/com/thebois/models/world/IWorld.java

```diff
@@ -35,4 +37,21 @@
      */
     ITile getTileAt(int posX, int posY);
 
+    /**
+     * Finds a vacant spot and returns it.
+     *
+     * @return The vacant tile.
+     */
+    ITile getRandomVacantSpot();
+
+    /**
+     * Finds an empty neighbour of a tile that is closest to the given position.
+     *
+     * @param tile The tile to get neighbours of.
+     * @param from The position that needs to be closest to the returned neighbour.
+     *
+     * @return The position of the neighbouring tile that is the closest to the given position.
+     */
+    Optional<Position> getClosestNeighbourOf(ITile tile, Position from);
```

Comment:
Do we want to mention that if no neighbour was vacant, then the optional will be empty or is that understood by having an optional? 
Cause that is what I'm assuming the method should do, but from reading the Javadoc I feel like it returns the closest neighbour even if it's occupied.

What I'm trying to say is that we should mention in the Javadoc that it will give the closestVacantNeighbour.

Reply:
You're right, the fact that it returns an `Optional<Position>` implies that it might not return anything. But you're also right that it's good to be really clear in the javadoc about that behaviour as well, so the reader doesn't have to infer anything. 

Updating it.